### DISCUSSION TITLE
How can I make a dynamic route start with an @ symbol? Like "mywebsite.com/@username"

### DISCUSSION THREAD

**Discussion by lumenwrites** [vercel/next.js#31932] [Help] [answered]
My goal is to have two kinds of dynamic routes on the same path. I want:

- `mywebsite.com/@username` to send people to `/pages/[username].tsx`
- `mywebsite.com/page-slug` to send people to `/pages/[pageSlug].tsx`

I tried creating two files:

- `/pages/[pageSlug].tsx`
- `/pages/[...username].tsx`

So that `[pageSlug].tsx` would have priority, and then everything else would go to `[...username].tsx`, where I would extract the `username` from the path and render the page.

It works great locally, but on vercel `mywebsite.com/@username` pages return 404 error.

---

I have also tried following [this](https://stackoverflow.com/questions/66389057/adding-symbol-eg-in-front-of-slug-in-nextjs-route) advice about doing this with url rewrites, but it doesn't seem to work.

I have created two files:

- `/pages/[pageSlug].tsx`
- `/pages/user/[username].tsx`

Set up the rewrites in `next.config.js`:

    module.exports = {
      reactStrictMode: true,
      async rewrites() {
        return [
          {
            source: '/:username(@[a-zA-Z0-9]+)/:id*',
            destination: "/user/:username/:id*",
          }
        ]
      }
    }

And set up the links to the user profiles like so:

    <Link href="/user/[username]" as={`@${post.author.username}`}>
      {post.author.username}
    </Link>
    
Just loading the `http://localhost:3080/@lumen` works, but when I click on a link I'm getting:

> Error: The provided `as` value (/@lumen) is incompatible with the `href` value (/user/[username]). Read more: https://nextjs.org/docs/messages/incompatible-href-as

How can I make this work with the rewrites (or in any other way)? Any advice?

**Comment by wwwdepot** [comment, marked as answer]
Rewrite config

```js
  rewrites: async () => {
    return [
      {
        source: "/@:username",
        destination: "/user/:username",
      },
    ];
  },
```

Navigate to user page using Link component

```jsx
//     /page/user/[username].js
import Link from "next/link";

export default function UserPage({ user }) {
  return (
    <>
      <Link href="/@jack">
        <a>Jack</a>
      </Link>
      {" - "}
      <Link href="/@dave">
        <a>Dave</a>
      </Link>
      {" - "}
      <Link href="/@victor">
        <a>Victor</a>
      </Link>
      <pre>{JSON.stringify(user, null, 4)}</pre>
    </>
  );
}

export function getServerSideProps(ctx) {
  return {
    props: {
      user: ctx.params,
    },
  };
}
```

**Reply by lumenwrites** [reply]
This works! Thank you so much for your help!
